be/src/pipeline/pipeline_task.cpp

```diff
@@ -290,7 +290,7 @@ Status PipelineTask::execute(bool* eos) {
         return Status::OK();
     }
     // The status must be runnable
-    if (!_opened) {
+    if (!_opened && !_fragment_context->is_canceled()) {
```

Comment:
It will not fix this bug. If at line 293, query is not cancelled, you will go to line 294 and query is cancelled, it will core during line 294.